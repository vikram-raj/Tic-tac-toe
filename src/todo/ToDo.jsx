import React  from 'react';
import PropTypes  from 'prop-types';

export default function ToDo({complete, deleteToDo, text, toggleCompletion}) {
  return (
    <div className="to-do">
      <div className={complete ? 'to-do__info--complete' : 'to-do__info'}>
        <input
          checked={complete}
          className="to-do__completion"
          onChange={toggleCompletion}
          type="checkbox"
        />
        <span className="to-do__title">{text}</span>
      </div>
      <button onClick={deleteToDo}>X</button>
    </div>
  );
}

ToDo.propTypes = {
  complete: PropTypes.bool.isRequired,
  deleteToDo: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
  toggleCompletion: PropTypes.func.isRequired,
};