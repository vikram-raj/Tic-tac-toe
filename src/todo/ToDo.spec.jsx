import React from 'react';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import ToDo from './ToDo';

configure({adapter: new Adapter()});

describe('ToDo', () => {
  it('displays text based on props.text', () => {
    const toDo = shallow(
      <ToDo
        complete={false}
        deleteToDo={() => null}
        text="Test ToDo"
        toggleCompletion={() => null}
      />,
    );

    expect(toDo.find('.to-do__title').text()).toBe('Test ToDo');

    toDo.setProps({text: 'New ToDo'});

    expect(toDo.find('.to-do__title').text()).toBe('New ToDo');
  });

  xit('changes class based on props.completion', () => {
    const toDo = shallow(
      <ToDo
        complete
        deleteToDo={() => null}
        text="Test ToDo"
        toggleCompletion={() => null}
      />,
    );
    const inCompleteToDo = shallow(
      <ToDo
        complete={false}
        deleteToDo={() => null}
        text="Test ToDo"
        toggleCompletion={() => null}
      />,
    );

    // The first time this test is run, it will generate a snapshot of the component's output.
    // As further changes are made to the component, you simply review the diff to ensure
    // that the changes are intentional.
    expect(toJson(toDo)).toMatchSnapshot();

    // We still need to check both potential states of the component!
    expect(toJson(inCompleteToDo)).toMatchSnapshot();
  });

  xit('ToDo calls props.toggleCompletion on checkbox change', () => {
    // Creating a sinon spy,
    // essentially a dummy function who's whole purpose is to tell us information about how it is called.
    const toggleCompletionSpy = jest.fn();

    // We pass the spy as a prop to the component
    const toDo = shallow(
      <ToDo
        complete
        deleteToDo={() => null}
        text="Test ToDo"
        toggleCompletion={toggleCompletionSpy}
      />,
    );

    toDo.find('.to-do__completion').simulate('change');

    expect(toggleCompletionSpy).toHaveBeenCalled();
  });

  xit('ToDo calls props.deleteTodo on delete button click', () => {
    const deleteToDoSpy = jest.fn();

    const toDo = shallow(
      <ToDo
        complete
        deleteToDo={deleteToDoSpy}
        text="Test ToDo"
        toggleCompletion={() => null}
      />,
    );

    toDo.find('button').simulate('click');

    expect(deleteToDoSpy).toHaveBeenCalled();
  });
});