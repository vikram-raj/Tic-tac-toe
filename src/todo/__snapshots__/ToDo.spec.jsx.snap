// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDo changes class based on props.completion 1`] = `
<div
  className="to-do"
>
  <div
    className="to-do__info--complete"
  >
    <input
      checked={true}
      className="to-do__completion"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="to-do__title"
    >
      Test ToDo
    </span>
  </div>
  <button
    onClick={[Function]}
  >
    X
  </button>
</div>
`;

exports[`ToDo changes class based on props.completion 2`] = `
<div
  className="to-do"
>
  <div
    className="to-do__info"
  >
    <input
      checked={false}
      className="to-do__completion"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="to-do__title"
    >
      Test ToDo
    </span>
  </div>
  <button
    onClick={[Function]}
  >
    X
  </button>
</div>
`;
